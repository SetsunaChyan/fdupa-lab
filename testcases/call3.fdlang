function main1(){
    a = 20;
    call callee1(a);
}

function callee1(m){
    m = m + 1;
    check_interval(m, 21, 21);
    call callee2(m);
}

function callee2(m){
    m = m - 1;
    n = 5;

    while (m > 10) {
        m = n + m;
        n = n + 1;
    }

    check_interval(m, 21, 21);
   
}

function main2(){
    a = 20;
    call callee3(a);
}

function callee3(m){
    m = m + 1;
    check_interval(m, 21, 21);
}
