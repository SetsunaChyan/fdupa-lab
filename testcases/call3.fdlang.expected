Line 8: YES; m in [21, 21]
Line 21: Unreachable
Line 32: YES; m in [21, 21]